[RFC] First draft for ALTC v2

ansor4 opened the pull request:
First draft for the format of ALTC v2, based off of what we discussed in earlier brainstorm sessions. Eager to get feedback before making this an official RFC!

erikdstock commented:
Nice. 2 thoughts: 

1. I think we should add a list of core competencies that we are teaching towards, with the goal that mentors and learners should be targeting these as they build curriculum. I can think of 2 batches:
- Fundamentals: Basically what we covered in the udacity tutorial, syntax, data types, functions, etc.
- How software/websites are made:
  - Students know what is an algorithm, an object, a class, etc
  - Students can talk about their code or how they might approach a problem (in pseudocode)
  - Students know what a package is (some familiarity with package manager for their language etc)
  - Students have a basic understanding of how their language connects to the internet or an application, eg what a server/client is, SQL (if it's relevant to their job) etc
 
2. On this note I think we should do some one-off workshops on JS, HTML and CSS (maybe markdown as well), things that anyone can come to with a computer, follow along and end up making something that looks like a website- If done through glitch it could be a github pages CV.

ashfurrow commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,96 @@
+- Build empathy across Artsy
+- It's fun! Boosts morale
+- Empowers non-technical people
+- Gets engineers used to active teaching for non-technical people
```
```suggestion
- Gets engineers used to teaching regularly
```

ashfurrow commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,96 @@
+
+## Motivations for ALTC v2
+
+- It's social! Lets Artsy people interact across their normal silos
```
Definitely – this is more and more important as we grow, and especially since sup's are not as actively attended anymore.

ansor4 commented:
👍 Thanks for the comments! I've added some more lines on mentor expectations. I've added a list of core competencies we could strive for, but don't want to limit or pressure groups to have to all of these covered. I don't have a good idea of what we should expect, so I'm just a bit wary on setting expectations, especially if it comes at the cost of engagement. What do you two think? This could be a bigger discussion too if we feel like we're ready to pull in mentors.

sepans commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+| Sunglass (capturing data and displaying it)      | https://editor.p5js.org/sepans/sketches/s9tmMft3R |
+| Load artwork image from artsy (API)              | https://editor.p5js.org/sepans/sketches/NbuDfgC47 |
+
+Here are tools that [@sepans][sepans] recommended to his group in the previous ALTC:
```
Maybe I wasn't clear about the purpose if these tools. These are the ones for those who wanted to move up and use a more general purpose editor instead of p5 editor otherwise p5 editor more or less support these. So we can move this list to the bottom of the doc only for those who want to use something other than p5 (or drop it if it is no longer relevant).

sepans commented:
👋 @shiffman @lmccart @therewasaguy we are going to kick off the second round of Artsy Learns to Code (ALTC) initiative (Artsy engineers helping colleagues learn coding) and based on the feedback from the first ALTC we are going to focus on `p5.js` this time. (tytyty for all the tools and tutorials 🙏 ❤️ )

If you have any comments, feedback or other resources in mind we would love to include them in this RFC. (also feel free to tag others here)

sweir27 commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+  - Loops
+  - Arrays
+  - Objects
+- Web Development:
```
Might be too in the weeds, but sections of https://github.com/alex/what-happens-when may also be relevant here

sweir27 commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+
+## Expectations of Learners
+
+- Passion to learn and try something new
```
How much time is expected of learners?

mdole replied:
+1 good to set an expectation. We could also consider having different "levels" of commitment. Some people will probably want to do < 1 hr/week, while others might want to do 5+. Having a group looking for a similar level of commitment and a mentor prepared to spend more time (or perhaps just a more experienced mentor) could be really key in keeping folks engaged and interested

ansor4 replied:
That's a great idea! My initial thought was to not set any expectation for commitments and have each group work out how often they'd like to meet. But setting up groups with similar expectations to at the get go will be a great kickoff point.

sweir27 commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+
+## Expectations of Mentoring Engineers
+
+- Mentors should expect to spend 1-2 hrs a week of their time to meet regularly and answer questions async on slack
```
For how many weeks?

(For context, trying to balance this time commitment against other time commitments-- like mentoring new engineers! It might be helpful to give a ballpark here.)

ansor4 replied:
Great idea, I'll put a ballpark similar to the last ALTC, if not a little shorter. I feel some learners lost some steam near the end, but I'll be sure to note that actual timeline will be at mentor's discretion.

lmccart commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+| Resource Name                                            | Link                                          |
+| -------------------------------------------------------- | --------------------------------------------- |
+| Cool p5 sketches to tweak and play with                  | http://www.generative-gestaltung.de/2/        |
+| p5 DOM: use p5 to create and interact with HTML elements | https://p5js.org/reference/#/libraries/p5.dom |
```
p5.dom is now merged into core, and is included in main reference page. no need to direct to a separate link :)

lmccart commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+| p5 editor                                                | https://editor.p5js.org/                      |
+| p5 video tutorials                                       | https://www.youtube.com/user/shiffman/        |
+| Tutorial examples                                        | https://p5js.org/examples/                    |
+| Reference                                                | https://p5js.org/reference/                   |
```
http://openprocessing.org/
huge collection of sketches made with p5, with viewable code

mdole commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+- Build empathy across Artsy
+- It's fun! Boosts morale
+- Empowers non-technical people
+- Gets engineers used to teaching regularly
```
This was especially huge for me last time - I definitely was challenged to understand the tools I use on a daily basis better because I was teaching them to other folks who had the annoying habit of asking good questions

mdole commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+## Structure
+
+Similar to the previous ALTC, we'll have trios: one mentor and two learners, who will be partners throughout _Artsy
+Learns to Code_.
```
Last time we immediately had to expand to 4 learners per mentor. Should we just build that in this time around?

ansor4 replied:
Hmm, maybe let's gauge interest and see? I really like the idea of smaller more intimate groups: fewer learners get left behind and the barrier for questions is lower.

mdole commented on events/artsy-learns-to-code-v2.md:
```diff
@@ -0,0 +1,116 @@
+
+Then groups can decide on what they might want to make with p5.js, and mentors can guide them.
+
+### Some Core Competencies We Could Aim Teach
```
missing a `to`

ansor4 commented:
Thanks for all the comments y'all! I'm gonna merge this PR as our starting point for ALTC v2!
